vagrant/ansible/roles/samba-glusterfs.setup/tasks/main.yml

```diff
@@ -0,0 +1,76 @@
+---
+    - name: Install samba and samba-vfs-glusterfs packages
+      yum:
+        name:
+          - samba
+          - samba-vfs-glusterfs
+        state: present
+
+    - name: Enable firewall for samba
+      firewalld:
+        service: samba
+        permanent: yes
+        state: enabled
+
+    - name: Install libsemanage-python. This is needed for the seboolean ansible command
+      yum: name=libsemanage-python state=present
+
+    - name: Selinux - Allow Samba to access glusterfs services
+      seboolean:
+        name: samba_load_libgfapi
+        state: yes
+        persistent: yes
+
+    - name: Copy over smb.conf
+      copy:
+        src: smb.conf
+        dest: /etc/samba/smb.conf
+        owner: root
+        group: root
+        mode: 0644
+
+    - name: Set Volume options and restart volume in a single host
+      block:
+      - name: Setup Gluster volume settings
+        gluster_volume:
+          state: present
+          name: "{{ gluster_cluster_volume }}"
+          options:
+            {
+              server.allow-insecure: 'on',
+              performance.cache-samba-metadata: 'on',
+              storage.batch-fsync-delay-usec: '0',
+              user.smb: 'on'
+            }
+      #Stop and Start gluster volume to run samba hooks to modify smb.conf
+      - name: Stop gluster volume
+        gluster_volume:
+          state: stopped
+          name: "{{ gluster_cluster_volume }}"
+      - name: Start gluster volume
+        gluster_volume:
+          state: started
+          name: "{{ gluster_cluster_volume }}"
+      run_once: true
+
+    - name: Create directory to mount share using fuse glusterfs
+      file:
+        path: /mnt-fuse
+        state: directory
+        mode: '0777'
+
+    - name: Mount gluster volume using glusterfs
+      mount:
+        path: /mnt-fuse
+        src: 'localhost:/vol'
+        fstype: glusterfs
+        state: mounted
+
+    - name: Set permission of share root to 0777
+      file:
+        path: /mnt-fuse
+        mode: '0777'
+      run_once: true
```

Comment:
and make sure that we unmount

Reply:
Why is this mounted at all? Only to set the perms?

Reply:
Anoop, Done.
Obnox, yes, we set the root as world writable to allow us to test easily.